### PR TITLE
Add keyboard shortcut to refresh feed

### LINKED ISSUE #23660: Use the ‘.’ key to refresh the feeds in slow mode
### Pitch

Other social media like twitter can refresh the timeline and move to the top by pressing the '.' key.  This is currently not possible on Mastodon sites. 

### Motivation

This is needed for mouseless browsing, which is faster and less RSI inducing.

### PR DESCRIPTION
**Pull request by TheDevJoao**
This PR resolves [#23660 ](https://github.com/mastodon/mastodon/issues/23660)

My solution adds a method called `handleHotkeyRefreshFeed` to the UI's `index` file, adding a keyboard shortcut  to the "." key. This method allows the user to scroll to the first article element in the page to better the overall experience of mouseless navigation.
 
#### Preview:
scrollToTop

#### Updated legend:
image

P.S: With my latest refactor the user can also scroll back to the top of the first toot/status of their home timeline in the advanced web interface's multi column mode but I'm not sure how to replicate this in other timelines: 
advancedScrollToTop

If anyone has a suggestion on how I can achieve this please let me know!

### CODE REVIEW

**Comment by tribela** [comment]
`window.scroll` would be a problem in multi column mode.
Would be nice to look number shortcut (that selects the column and go to top of that column)

**Comment by TheDevJoao** [comment]
> `window.scroll` would be a problem in multi column mode. Would be nice to look number shortcut (that selects the column and go to top of that column)

Hey @tribela! Thanks for pointing that out, I had  completely forgotten about multi columns in Mastodon 😆 
I recently refactored my code and partially solved this, check it out if you can! 🤝

**Comment by TheDevJoao** [comment]
Hey @ClearlyClaire could you review this whenever you have time? 😄

File: app/javascript/mastodon/locales/defaultMessages.json [hunk 1]
```diff
@@ -3236,7 +3240,7 @@
         "id": "onboarding.steps.follow_people.title"
       },
       {
-        "defaultMessage": "You curate your own feed. Let's fill it with interesting people.",
+        "defaultMessage": "You curate your own feed. Lets fill it with interesting people.",
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/locales/defaultMessages.json:3243]: This has wrongfully snuck in during a rebase and should be fixed!

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution!

I'm a bit confused, the description says it refreshes the feed, but what it does is focus the first post in the timeline? What am I missing here?

File: app/javascript/mastodon/locales/defaultMessages.json (hunk 1 above)
Comment (TheDevJoao) [reply app/javascript/mastodon/locales/defaultMessages.json:3243]: > 
Hey, @ClearlyClaire, thanks for the review!

I used "refresh feed/timeline" in most of the implementation due to the issue's request and saw that a little blue bird also calls this feature in a similar fashion, but after opening the PR and  @tribela's comment, I figured the best solution would be to focus on the first element in a `scrollable` section.

After analyzing the codebase more thoroughly, I realized that there is already an implementation that does this called `focusColumn` that utilizes the `'1', '2', '3', '4', '5', '6', '7', '8', '9'` keys, but this is only true if the advanced web interface option is enabled.

Maybe my solution would make more sense if `refresh timeline` and `refresh feed` were called something else. Any suggestions on potential names to refactor?

File: app/javascript/mastodon/features/ui/index.jsx [hunk 2]
```diff
@@ -485,6 +486,24 @@ class UI extends PureComponent {
...
+        if (firstArticle) {
+          firstArticle.scrollIntoView();
+        }
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
```
[the remark this answers is not in the document]
Comment (TheDevJoao) [reply app/javascript/mastodon/features/ui/index.jsx:503]: > I don't understand why you are using this pattern of registering an event handler on the fly: managing the event handler is `react-hotkeys`'s job.

You're right, I'll make sure to fix this in my next commit.

**Comment by mjankowski** [comment]
Closing due to lack of activity and merge conflicts -- that said, if you want to push this forward, can you rebase, and reopen, and then either a) update the description/names to reflect what the code actually does, or b) update the code to actually do the refresh that the desc/labels claim? (this seems more elaborate than what's there now, unless I'm missing something?)

**Pull request closed without merging**